### PR TITLE
PaketUnityInstall groovy only solution

### PR DESCRIPTION
**Pull request by marcolink**
## Description
To get more controllable and maintainable  sources, we now execute all paket unity operations directly via groovy, without the use of `paket.unity3d.exe`

## Changes
* ![CHANGE] Task :paketUnityInstall 
* ![ADD] A lot of Tests for :paketUnityInstall 
* ![ADD] respect `framework` property of `paket.dependencies` file

**the new `framework` property:**
starting with this changes we respect the optional property `framework` of paket dependencies.
more details on how it works can be found [here](https://fsprojects.github.io/Paket/dependencies-file.html#Framework-restrictions). When copying dependencies into the unity project, we respect this property for all `*.dll` files. 

If no framework restrictions are defined, we fallback to `net35` only. We might extend this in the future. for now, we're reflecting the [old behavior of paket.unity3d.exe](https://github.com/wooga/Paket.Unity3D/blob/master/src/Paket.Unity3D/Constants.fs#L10).

**Pro:**
- Maintainable source
- Gradle plugin
- extendable (support for github/http sources)

**Contra:**
- Gradle plugin (Gradle usage mandatory)

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy [hunk 1]
```diff
@@ -17,18 +17,128 @@
...
+        }
+
+        project.files(files)
+    }
+
+    Set<File> getFilesForPackage(String nuget) {
+        def fileTree = project.fileTree(dir: project.projectDir)
+        fileTree.include("packages/${nuget}/content/**")
+
+        frameworks.each({
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy]: use `getFrameworks()` to make it possible to set it via convention mapping.

File: src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy [hunk 2]
```diff
@@ -17,18 +17,128 @@
...
+                    parent = parent.parentFile
+                }
+
+                assert removed.removed
+                assert !outputPath.exists()
+            }
+        })
+    }
+
+    File transformInputToOutputPath(File inputFile, File baseDirectory) {
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/paket/unity/tasks/PaketUnityInstall.groovy]: mark this method `private` or `protected`

**Comment by doubleday** [comment]
No real review yet but just to give my 5c already: I like were this is going and I don't see a problem in making gradle mandatory because in my view it is already.

**Comment by Larusso** [comment]
Ok since nobody really cares about this I want to explain why we want some more buy in for this particular change. 

I created the first gradle plugins as a kind of extension to the current build tools used. The idea was to have a wrapper that can initialize and execute the toolchain in an easy way but with backwards compatibility in mind. Means you could still use your shell scripts etc.

This change would shift away from this approach. @marcolink implemented the new framework support already in the gradle only solution and we have no plans to update [Paket.Unity3D](https://wooga.github.io/Paket.Unity3D/). And we want to add more features. That will force any team with the desire to use a newer c# runtime support and external libraries or one of the new features we have in mind to shift to gradle as the main build tool. @doubleday already gave his blessing to move in this direction (even though he didn't accept this pr 🤣 ).

So are all ok with this direction?

**Comment by skrohn** [comment]
So what would I do instead of paket.unity3d install ? Or what is the proposed workflow to update/install packages?

**Comment by marcolink** [comment]
@skrohn use gradle for all operations related to dependency management.

**Comment by skrohn** [comment]
We are not using it so far. So I'd need some insights into the workflow at some point. I also don't see an issue with adding a dependency on gradle since everyone needs it for Android anyway. There should just be super easy instructions on how to setup your project for this. (there probably is, but i don't know where...)

**Comment by Larusso** [comment]
@skrohn We're working on the documentation part. You could look at my side branch in your repo.
run `./gradlew tasks` and you can see what it has to offer.

**Pull request merged**